proto/types.proto

```diff
@@ -0,0 +1,140 @@
+syntax = "proto3";
+package lazyledger.proto;
+
+import "google/protobuf/timestamp.proto";
+
+message Block {
+  Header header = 1;
+  AvailableDataHeader availableDataHeader = 2;
+  AvailableData availableData = 3;
+  Commit lastCommit = 4;
+}
+
+message Header {
+  uint64 version = 1;
+  uint64 chain_id = 2;
+  uint64 height = 3;
+  google.protobuf.Timestamp time = 4;
+  BlockID last_block_id = 5;
+  HashDigest last_commit_root = 6;
+  HashDigest consensus_hash = 7;
+  HashDigest state_commitment = 8;
+  HashDigest available_data_root = 9;
+  Address proposer_address = 10;
+}
+
+message AvailableDataHeader {
+  repeated HashDigest available_data_commitments = 1;
+}
+
+message AvailableData {
+  TransactionData transaction_data = 1;
+  IntermediateStateRootData intermediate_state_root_data = 2;
+  EvidenceData evidence_data = 3;
+  MessageData message_data = 4;
+}
+
+message Commit {
+  uint64 height = 1;
+  uint64 round = 2;
+  BlockID block_id = 3;
+  repeated CommitSig signatures = 4;
+}
+
+message BlockID {
+  HashDigest header_root = 1;
+}
+
+message HashDigest {
+  bytes raw_data = 1;
+}
+
+message Address {
+  bytes raw_data = 1;
+}
+
+message CommitSig {
+  enum BlockIDFlag {
+    BLOCK_ID_FLAG_UNSPECIFIED = 0;
```

Comment:
This is probably just copied and adapted from the tendermint protos but I'm wondering about the  the rationale behind this extra value. Shouldn't every value besides 1,2,3 unspecified/invalid?

I'll ping @marbar3778 about it.

Reply:
Ah NVM, this is just a proto detail:
> - There must be a zero value, so that we can use 0 as a numeric default value.
> - The zero value needs to be the first element, for compatibility with the proto2 semantics where the first enum value is always the default.

https://developers.google.com/protocol-buffers/docs/proto3#enum

Reply:
Yep, that was it.